storehaus-mysql/src/main/scala/com/twitter/storehaus/mysql/MySqlStore.scala

```diff
@@ -97,59 +100,68 @@ class MySqlStore(protected [mysql] val client: Client, table: String, kCol: Stri
   protected [mysql] def commitTransaction : Future[Unit] = client.query(COMMIT_TXN_SQL).unit
   protected [mysql] def rollbackTransaction : Future[Unit] = client.query(ROLLBACK_TXN_SQL).unit
 
-  protected [mysql] def executeMultiInsert[K1 <: MySqlValue](kvs: Map[K1, MySqlValue]) = {
-    val insertSql = MULTI_INSERT_SQL_PREFIX + Stream.continually("(?, ?)").take(kvs.size).mkString(",")
-    val insertParams = kvs.map { kv =>
-      List(MySqlStringInjection(kv._1).getBytes, MySqlStringInjection(kv._2).getBytes)
+  protected [mysql] def executeMultiInsert[K1 <: MySqlValue](
+      kvs: Map[K1, MySqlValue]): Future[Result] = {
+    val insertSql = MULTI_INSERT_SQL_PREFIX +
+      Stream.continually("(?, ?)").take(kvs.size).mkString(",")
+    val insertParams = kvs.map {
+      case (k, v) => List(k, v).map(String2MySqlValueInjection.invert(_).getOrElse("").getBytes)
     }.toSeq.flatten.map(Parameter.wrap[Array[Byte]])
-    client.prepare(insertSql)(insertParams:_*)
+    client.prepare(insertSql)(insertParams: _*)
   }
 
-  protected [mysql] def executeMultiUpdate[K1 <: MySqlValue](kvs: Map[K1, MySqlValue]) = {
-    val updateSql = MULTI_UPDATE_SQL_PREFIX + Stream.continually("WHEN ? THEN ?").take(kvs.size).mkString(" ") +
+  protected [mysql] def executeMultiUpdate[K1 <: MySqlValue](
+      kvs: Map[K1, MySqlValue]): Future[Result] = {
+    val updateSql = MULTI_UPDATE_SQL_PREFIX +
+      Stream.continually("WHEN ? THEN ?").take(kvs.size).mkString(" ") +
       MULTI_UPDATE_SQL_INFIX + Stream.continually("?").take(kvs.size).mkString("(", ",", ")")
 
     val updateParams = kvs.map { kv =>
-      (MySqlStringInjection(kv._1).getBytes, MySqlStringInjection(kv._2).getBytes)
+      (String2MySqlValueInjection.invert(kv._1).getOrElse("").getBytes,
+        String2MySqlValueInjection.invert(kv._2).getOrElse("").getBytes)
     }
     // params for "WHEN ? THEN ?"
-      val updateCaseParams: Seq[Parameter] =
-        updateParams.map { kv => List(kv._1, kv._2) }.toSeq.flatten.map(Parameter.wrap[Array[Byte]])
+    val updateCaseParams: Seq[Parameter] =
+      updateParams.map { kv => List(kv._1, kv._2) }.toSeq.flatten
```

Comment:
`flatMap { case (k, v) => List(k, v) }` I think we will work if you keep the return type annotation.